### PR TITLE
Slonopotamus.Stevedore version 0.3.3

### LINKED ISSUE #54478: Stevedore needs /norestart flag
Slonopotamus.Stevedore

Reference PR https://github.com/microsoft/winget-pkgs/pull/54370

---

For the reference, here's how Chocolatey installs MSI packages:

```
2022-03-15 17:48:57,518 12832 [DEBUG] - Running Start-ChocolateyProcessAsAdmin -validExitCodes '0 3010 1641' -workingDirectory 'C:\Users\m.radchenko.admin\AppData\Local\Temp\chocolatey\stevedore\0.3.3' -statements '/i "C:\Users\m.radchenko.admin\AppData\Local\Temp\chocolatey\stevedore\0.3.3\stevedore-0.3.3-x86_64.msi" /qn /norestart /l*v "C:\Users\m.radchenko.admin\AppData\Local\Temp\chocolatey\stevedore.0.3.3.MsiInstall.log" ' -exeToRun 'C:\windows\System32\msiexec.exe' 
2022-03-15 17:48:57,534 12832 [DEBUG] - Test-ProcessAdminRights: returning True
2022-03-15 17:48:57,942 12832 [DEBUG] - Elevating permissions and running ["C:\windows\System32\msiexec.exe" /i "C:\Users\m.radchenko.admin\AppData\Local\Temp\chocolatey\stevedore\0.3.3\stevedore-0.3.3-x86_64.msi" /qn /norestart /l*v "C:\Users\m.radchenko.admin\AppData\Local\Temp\chocolatey\stevedore.0.3.3.MsiInstall.log" ]. This may take a while, depending on the statements.
2022-03-15 17:49:11,408 12832 [DEBUG] - Command ["C:\windows\System32\msiexec.exe" /i "C:\Users\m.radchenko.admin\AppData\Local\Temp\chocolatey\stevedore\0.3.3\stevedore-0.3.3-x86_64.msi" /qn /norestart /l*v "C:\Users\m.radchenko.admin\AppData\Local\Temp\chocolatey\stevedore.0.3.3.MsiInstall.log" ] exited with '3010'.
2022-03-15 17:49:11,415 12832 [DEBUG] - Finishing 'Start-ChocolateyProcessAsAdmin'
2022-03-15 17:49:11,799 12832 [INFO ] - stevedore has been installed.
```

So, you may see that there's `/norestart` flag.

_Originally posted by @slonopotamus in https://github.com/microsoft/winget-cli/issues/229#issuecomment-1068076523_

### PR DESCRIPTION
**Pull request by slonopotamus**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/54370)

### CODE REVIEW

**Comment by slonopotamus** [comment]
I was unable to verify "Does your manifest conform to the 1.0 schema?" because there are no instructions on how I do that. On one hand, manifest was produced using `wingetcreate new`, so it is very likely valid. On the other hand, it contains `ManifestVersion: 1.1.0`. Does 1.1.0 version conform to 1.0 schema? I have no idea.

Another issue: I've tested `winget install --manifest <path>` and it does install the package successfully. But it causes immediate reboot. I came across https://github.com/microsoft/winget-cli/issues/229 but that doesn't help very much. Stevedore MSI is **not forcing** a reboot. It only [**schedules** a reboot](https://github.com/slonopotamus/stevedore/blob/0.3.3/wix/main.wxs#L296). If you install Stevedore MSI manually, you're presented with a "would you like to reboot" dialog. If you install Stevedore via [Chocolatey package](https://community.chocolatey.org/packages/stevedore), it returns 3010 exit code and Chocolatey warns user that reboot is required. So, I am not sure what and why transforms **scheduled** reboot into a **forced** reboot. But I believe this is something that users are unhappy about in https://github.com/microsoft/winget-cli/issues/229 and what needs to be fixed on WinGet side.

**Review by ItzLevvie — CHANGES_REQUESTED** [review]
Since validation is done on VMs I don't know if this would go through because I'm stuck here:
```
WixQuietExec:  Get-LocalGroup : Group docker-users was not found.
WixQuietExec:  At line:1 char:2
WixQuietExec:  + (Get-LocalGroup -Name docker-users) -or (New-LocalGroup -Name docker- ...
WixQuietExec:  +  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
WixQuietExec:      + CategoryInfo          : ObjectNotFound: (docker-users:String) , GroupNotFoundException
WixQuietExec:      + FullyQualifiedErrorId : GroupNotFound,Microsoft.PowerShell.Commands.GetLocalGroupCommand
WixQuietExec:   
WixQuietExec:  True
```

and the installer stalls

File: manifests/s/Slonopotamus/Stevedore/0.3.3/Slonopotamus.Stevedore.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Created using wingetcreate 0.5.0.1
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.1.0.schema.json
+
+PackageIdentifier: Slonopotamus.Stevedore
+PackageVersion: 0.3.3
+Installers:
```
Comment (OfficialEsco) [inline manifests/s/Slonopotamus/Stevedore/0.3.3/Slonopotamus.Stevedore.installer.yaml:6]:
```suggestion
InstallerSwitches:
  Custom: /norestart
  Upgrade: /norestart
Installers:
```

**Review by OfficialEsco — CHANGES_REQUESTED** [review]

**Review by OfficialEsco — DISMISSED** [review]

**Comment by OfficialEsco** [comment]
<details>
<summary>Click for Log</summary>
<br>

```
=== Logging started: 3/18/2022  17:02:20 ===
Action start 17:02:20: INSTALL.
Action start 17:02:20: FindRelatedProducts.
Action ended 17:02:20: FindRelatedProducts. Return value 1.
Action start 17:02:20: LaunchConditions.
Action ended 17:02:20: LaunchConditions. Return value 1.
Action start 17:02:20: ValidateProductID.
Action ended 17:02:20: ValidateProductID. Return value 1.
Action start 17:02:20: CostInitialize.
Action ended 17:02:20: CostInitialize. Return value 1.
Action start 17:02:20: FileCost.
Action ended 17:02:20: FileCost. Return value 1.
Action start 17:02:20: CostFinalize.
Action ended 17:02:20: CostFinalize. Return value 1.
Action start 17:02:20: SetARPINSTALLLOCATION.
Action ended 17:02:20: SetARPINSTALLLOCATION. Return value 1.
Action start 17:02:20: MigrateFeatureStates.
Action ended 17:02:20: MigrateFeatureStates. Return value 0.
Action start 17:02:20: InstallValidate.
Action ended 17:02:20: InstallValidate. Return value 1.
Action start 17:02:20: InstallInitialize.
Action ended 17:02:20: InstallInitialize. Return value 1.
Action start 17:02:20: RemoveExistingProducts.
Action ended 17:02:20: RemoveExistingProducts. Return value 1.
Action start 17:02:20: ProcessComponents.
Action ended 17:02:23: ProcessComponents. Return value 1.
Action start 17:02:23: UnpublishFeatures.
Action ended 17:02:23: UnpublishFeatures. Return value 1.
Action start 17:02:23: StopServices.
Action ended 17:02:23: StopServices. Return value 1.
Action start 17:02:23: DeleteServices.
Action ended 17:02:23: DeleteServices. Return value 1.
Action start 17:02:23: RemoveRegistryValues.
Action ended 17:02:23: RemoveRegistryValues. Return value 1.
Action start 17:02:23: RemoveShortcuts.
Action ended 17:02:23: RemoveShortcuts. Return value 1.
Action start 17:02:23: RemoveEnvironmentStrings.
Action ended 17:02:23: RemoveEnvironmentStrings. Return value 1.
Action start 17:02:23: RemoveFiles.
Action ended 17:02:23: RemoveFiles. Return value 0.
Action start 17:02:23: SetCreateDockerGroup.
Action ended 17:02:23: SetCreateDockerGroup. Return value 1.
Action start 17:02:23: CreateDockerGroup.
Action ended 17:02:23: CreateDockerGroup. Return value 1.
Action start 17:02:23: SetInstallContainersFeature.
Action ended 17:02:23: SetInstallContainersFeature. Return value 1.
Action start 17:02:23: InstallContainersFeature.
Action ended 17:02:23: InstallContainersFeature. Return value 1.
Action start 17:02:23: SetInstallHyperVFeature.
Action ended 17:02:23: SetInstallHyperVFeature. Return value 1.
Action start 17:02:23: InstallHyperVFeature.
Action ended 17:02:23: InstallHyperVFeature. Return value 1.
Action start 17:02:23: SetInstallWSL2Feature.
Action ended 17:02:23: SetInstallWSL2Feature. Return value 1.
Action start 17:02:23: InstallWSL2Feature.
Action ended 17:02:23: InstallWSL2Feature. Return value 1.
Action start 17:02:23: ConfigureUsers.
Action start 17:02:26: CreateUserRollback.
Action ended 17:02:26: CreateUserRollback. Return value 1.
Action start 17:02:26: CreateUser.
Action ended 17:02:26: CreateUser. Return value 1.
Action ended 17:02:26: ConfigureUsers. Return value 1.
Action start 17:02:26: InstallFiles.
Action ended 17:02:26: InstallFiles. Return value 1.
Action start 17:02:26: CreateShortcuts.
Action ended 17:02:26: CreateShortcuts. Return value 1.
Action start 17:02:26: WriteRegistryValues.
Action ended 17:02:26: WriteRegistryValues. Return value 1.
Action start 17:02:26: WriteEnvironmentStrings.
Action ended 17:02:26: WriteEnvironmentStrings. Return value 1.
Action start 17:02:26: InstallServices.
Action ended 17:02:26: InstallServices. Return value 1.
Action start 17:02:26: SchedServiceConfig.
Action start 17:02:26: RollbackServiceConfig.
Action ended 17:02:26: RollbackServiceConfig. Return value 1.
Action start 17:02:26: ExecServiceConfig.
Action ended 17:02:26: ExecServiceConfig. Return value 1.
Action ended 17:02:26: SchedServiceConfig. Return value 1.
Action start 17:02:26: StartServices.
Action ended 17:02:26: StartServices. Return value 1.
Action start 17:02:26: RegisterUser.
Action ended 17:02:26: RegisterUser. Return value 1.
Action start 17:02:26: RegisterProduct.
Action ended 17:02:26: RegisterProduct. Return value 1.
Action start 17:02:26: PublishFeatures.
Action ended 17:02:26: PublishFeatures. Return value 1.
Action start 17:02:26: PublishProduct.
Action ended 17:02:26: PublishProduct. Return value 1.
Action start 17:02:26: InstallFinalize.
WixQuietExec:  Get-LocalGroup : Group docker-users was not found.
WixQuietExec:  At line:1 char:2
WixQuietExec:  + (Get-LocalGroup -Name docker-users) -or (New-LocalGroup -Name docker- ...
WixQuietExec:  +  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
WixQuietExec:      + CategoryInfo          : ObjectNotFound: (docker-users:String) , GroupNotFoundException
WixQuietExec:      + FullyQualifiedErrorId : GroupNotFound,Microsoft.PowerShell.Commands.GetLocalGroupCommand
WixQuietExec:   
WixQuietExec:  True
WixQuietExec:  WARNING: Restart is suppressed because NoRestart is specified.
WixQuietExec:  
WixQuietExec:  
WixQuietExec:  Path          : 
WixQuietExec:  Online        : True
WixQuietExec:  RestartNeeded : True
WixQuietExec:  
WixQuietExec:  
WixQuietExec:  
WixQuietExec:  WARNING: Restart is suppressed because NoRestart is specified.
WixQuietExec:  
WixQuietExec:  
WixQuietExec:  Path          : 
WixQuietExec:  Online        : True
WixQuietExec:  RestartNeeded : True
WixQuietExec:  
WixQuietExec:  
WixQuietExec:  
WixQuietExec:  WARNING: Restart is suppressed because NoRestart is specified.
WixQuietExec:  
WixQuietExec:  
WixQuietExec:  Path          : 
WixQuietExec:  Online        : True
WixQuietExec:  RestartNeeded : True
WixQuietExec:  
WixQuietExec:  
WixQuietExec:  
Action ended 17:03:18: InstallFinalize. Return value 1.
Action start 17:03:18: ScheduleReboot.
Action ended 17:03:18: ScheduleReboot. Return value 1.
Action ended 17:03:18: INSTALL. Return value 1.
Property(S): DiskPrompt = Stevedore Installation
Property(S): UpgradeCode = {9A68EC9C-B494-4CDD-B081-F074FE91BF0B}
Property(S): InstallContainersFeature = "C:\Windows\sysnative\WindowsPowerShell\v1.0\powershell.exe" -Command "Enable-WindowsOptionalFeature -Online -FeatureName Containers -All -NoRestart"
Property(S): InstallHyperVFeature = "C:\Windows\sysnative\WindowsPowerShell\v1.0\powershell.exe" -Command "Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V -All -NoRestart"
Property(S): InstallWSL2Feature = "C:\Windows\sysnative\WindowsPowerShell\v1.0\powershell.exe" -Command "Enable-WindowsOptionalFeature -Online -FeatureName 'Microsoft-Windows-Subsystem-Linux;VirtualMachinePlatform' -All -NoRestart"
Property(S): INSTALLDIR = C:\Program Files\Stevedore\
Property(S): bin = C:\Program Files\Stevedore\bin\
Property(S): cli_plugins = C:\Program Files\Docker\cli-plugins\
Property(S): WixUIRMOption = UseRM
Property(S): ALLUSERS = 1
Property(S): CreateDockerGroup = "C:\Windows\sysnative\WindowsPowerShell\v1.0\powershell.exe" -Command "(Get-LocalGroup -Name docker-users) -or (New-LocalGroup -Name docker-users -Description 'Users of Docker Desktop')"
Property(S): ARPINSTALLLOCATION = C:\Program Files\Stevedore\
Property(S): CreateUser = **********
Property(S): CreateUserRollback = **********
Property(S): ExecServiceConfig = SchedServiceConfigstevedore1restartrestartnone160
Property(S): RollbackServiceConfig = SchedServiceConfig
Property(S): ProgramFiles64Folder = C:\Program Files\
Property(S): Docker = C:\Program Files\Docker\
Property(S): TARGETDIR = D:\
Property(S): ProgramMenuFolder = C:\ProgramData\Microsoft\Windows\Start Menu\Programs\
Property(S): DesktopFolder = C:\Users\Public\Desktop\
Property(S): SourceDir = C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\
Property(S): VersionNT = 603
Property(S): SERVICENAME = stevedore
Property(S): DISABLEADVTSHORTCUTS = 1
Property(S): POWERSHELLEXE = C:\Windows\sysnative\WindowsPowerShell\v1.0\powershell.exe
Property(S): ARPPRODUCTICON = stevedore.ico
Property(S): ARPHELPLINK = https://github.com/slonopotamus/stevedore
Property(S): Manufacturer = Marat Radchenko
Property(S): ProductCode = {4E2D6003-5D2F-444C-B957-6320DB29343F}
Property(S): ProductLanguage = 1033
Property(S): ProductName = Stevedore
Property(S): ProductVersion = 0.3.3
Property(S): DefaultUIFont = WixUI_Font_Normal
Property(S): WixUI_Mode = FeatureTree
Property(S): ErrorDialog = ErrorDlg
Property(S): SecureCustomProperties = WIX_DOWNGRADE_DETECTED;WIX_UPGRADE_DETECTED
Property(S): MsiHiddenProperties = CreateUser;CreateUserRollback;RemoveUser
Property(S): MsiLogFileLocation = C:\Users\validator\AppData\Local\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir\WinGet-Slonopotamus.Stevedore.0.3.3-2022-03-18-17-02-16.775.log
Property(S): PackageCode = {A8434072-F7E0-4400-BB0A-77D7D4BC2343}
Property(S): ProductState = -1
Property(S): PackagecodeChanging = 1
Property(S): REBOOTPROMPT = S
Property(S): REBOOT = ReallySuppress
Property(S): CURRENTDIRECTORY = C:\Users\validator\AppData\Local\Microsoft\WindowsApps
Property(S): CLIENTUILEVEL = 2
Property(S): CLIENTPROCESSID = 5792
Property(S): VersionDatabase = 500
Property(S): VersionMsi = 5.00
Property(S): VersionNT64 = 603
Property(S): WindowsBuild = 9600
Property(S): ServicePackLevel = 0
Property(S): ServicePackLevelMinor = 0
Property(S): MsiNTProductType = 1
Property(S): WindowsFolder = C:\Windows\
Property(S): WindowsVolume = C:\
Property(S): System64Folder = C:\Windows\system32\
Property(S): SystemFolder = C:\Windows\SysWOW64\
Property(S): RemoteAdminTS = 1
Property(S): TempFolder = C:\Users\VALIDA~1\AppData\Local\Temp\
Property(S): ProgramFilesFolder = C:\Program Files (x86)\
Property(S): CommonFilesFolder = C:\Program Files (x86)\Common Files\
Property(S): CommonFiles64Folder = C:\Program Files\Common Files\
Property(S): AppDataFolder = C:\Users\validator\AppData\Roaming\
Property(S): FavoritesFolder = C:\Users\validator\Favorites\
Property(S): NetHoodFolder = C:\Users\validator\AppData\Roaming\Microsoft\Windows\Network Shortcuts\
Property(S): PersonalFolder = C:\Users\validator\Documents\
Property(S): PrintHoodFolder = C:\Users\validator\AppData\Roaming\Microsoft\Windows\Printer Shortcuts\
Property(S): RecentFolder = C:\Users\validator\AppData\Roaming\Microsoft\Windows\Recent\
Property(S): SendToFolder = C:\Users\validator\AppData\Roaming\Microsoft\Windows\SendTo\
Property(S): TemplateFolder = C:\ProgramData\Microsoft\Windows\Templates\
Property(S): CommonAppDataFolder = C:\ProgramData\
Property(S): LocalAppDataFolder = C:\Users\validator\AppData\Local\
Property(S): MyPicturesFolder = C:\Users\validator\Pictures\
Property(S): AdminToolsFolder = C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Administrative Tools\
Property(S): StartupFolder = C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Startup\
Property(S): StartMenuFolder = C:\ProgramData\Microsoft\Windows\Start Menu\
Property(S): FontsFolder = C:\Windows\Fonts\
Property(S): GPTSupport = 1
Property(S): OLEAdvtSupport = 1
Property(S): ShellAdvtSupport = 1
Property(S): MsiAMD64 = 6
Property(S): Msix64 = 6
Property(S): Intel = 6
Property(S): PhysicalMemory = 4096
Property(S): VirtualMemory = 3331
Property(S): AdminUser = 1
Property(S): MsiTrueAdminUser = 1
Property(S): LogonUser = validator
Property(S): UserSID = S-1-5-21-550170491-1932705404-2414133025-1001
Property(S): UserLanguageID = 1033
Property(S): ComputerName = DAAS20H2
Property(S): SystemLanguageID = 1033
Property(S): ScreenX = 1024
Property(S): ScreenY = 768
Property(S): CaptionHeight = 19
Property(S): BorderTop = 1
Property(S): BorderSide = 1
Property(S): TextHeight = 16
Property(S): TextInternalLeading = 3
Property(S): ColorBits = 32
Property(S): TTCSupport = 1
Property(S): Time = 17:03:18
Property(S): Date = 3/18/2022
Property(S): MsiNetAssemblySupport = 4.8.4084.0
Property(S): MsiWin32AssemblySupport = 6.3.19041.546
Property(S): RedirectedDllSupport = 2
Property(S): MsiRunningElevated = 1
Property(S): Privileged = 1
Property(S): USERNAME = validator
Property(S): DATABASE = C:\Windows\Installer\1fc5c.msi
Property(S): OriginalDatabase = C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\Slonopotamus.Stevedore.0.3.3.msi
Property(S): UILevel = 3
Property(S): MsiUIProgressOnly = 1
Property(S): MsiUIHideCancel = 1
Property(S): ACTION = INSTALL
Property(S): ROOTDRIVE = D:\
Property(S): CostingComplete = 1
Property(S): OutOfDiskSpace = 0
Property(S): OutOfNoRbDiskSpace = 0
Property(S): PrimaryVolumeSpaceAvailable = 0
Property(S): PrimaryVolumeSpaceRequired = 0
Property(S): PrimaryVolumeSpaceRemaining = 0
Property(S): INSTALLLEVEL = 1
Property(S): SOURCEDIR = C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\
Property(S): SourcedirProduct = {4E2D6003-5D2F-444C-B957-6320DB29343F}
Property(S): MsiRebootActionScheduled = 1
Property(S): ProductToBeRegistered = 1
MSI (s) (A4:2C) [17:03:18:147]: Product: Stevedore -- Installation completed successfully.

MSI (s) (A4:2C) [17:03:18:147]: Windows Installer installed the product. Product Name: Stevedore. Product Version: 0.3.3. Product Language: 1033. Manufacturer: Marat Radchenko. Installation success or error status: 0.

MSI (s) (A4:2C) [17:03:18:147]: Windows Installer requires a system restart. Product Name: Stevedore. Product Version: 0.3.3. Product Language: 1033. Manufacturer: Marat Radchenko. Type of System Restart: 2. Reason for Restart: 2.

MSI (s) (A4:2C) [17:03:18:147]: Product: Stevedore. Restart required. The installation or update for the product required a restart for all changes to take effect.  The restart was deferred to a later time.

=== Logging stopped: 3/18/2022  17:03:18 ===


```

</details>

This probably requires `Dependencies` -> `WindowsFeatures` which is not supported yet

**Review by ItzLevvie — APPROVED** [review]

**Pull request merged**